src/python/pants/backend/go/target_types.py

```diff
@@ -20,6 +20,7 @@ class GoSources(Sources):
 
 
 class GoPackageSources(GoSources):
+    alias = "_sources"
     default = ("*.go",)
```

Comment:
Tangential: In the future, the default should probably encompass all files in the directory since Go allows non-`.go` files in packages, for example, `.m` assembly files.

Reply:
I'll open an issue. But nothing we need to solve for the initial release.

Reply:
I imagine we'll want to enumerate suffixes, and not take literally all files in the directory, which might include weird tmpfiles or something. I have no idea what that would lead to...

Reply:
This seems more than tangential... does making the field private mean that you _can't_ set it? Because that would block those use cases right now.

Reply:
> I imagine we'll want to enumerate suffixes, and not take literally all files in the directory, which might include weird tmpfiles or something. I have no idea what that would lead to...

The `go` tool will include all files other than those that start with `_` or `.`.

Reply:
The glob could probably be `("*", "!_*", "!.*", "!BUILD*")`.

Reply:
That seems reasonable. So maybe we should do that now, to address the issue Stu raises?

Reply:
I agree, probably worth doing now. In any event, opened opened https://github.com/pantsbuild/pants/issues/12825 for the globs issue.

Reply:
Will wait to merge this PR until I have time to better figure out #12825. This isn't critical to land until we're ready to stabilize